simtools/testing/configuration.py

```diff
@@ -0,0 +1,225 @@
+                {
+                    "APPLICATION": app,
+                    "TEST_NAME": "auto-version",
+                    "CONFIGURATION": {"VERSION": True},
+                },
+                {"APPLICATION": app, "TEST_NAME": "auto-no_config"},
+            ]
+        )
+
+    if get_test_names:
+        return [
+            f"{item.get('APPLICATION', 'no-app-name')}_{item.get('TEST_NAME', 'no-test-name')}"
+            for item in configs
+        ]
+
+    return configs
+
+
+def _read_configs_from_files(config_files):
+    """Read test configuration from files."""
+    configs = []
+    for config_file in config_files:
+        # read config file
+        # remove new line characters from config - otherwise issues
+        # with especially long file names
+        _dict = gen.remove_substring_recursively_from_dict(
+            gen.collect_data_from_file_or_dict(file_name=config_file, in_dict=None), substring="\n"
+        )
+        configs.append(_dict.get("CTA_SIMPIPE", None))
+    return configs
+
+
+def configure(config, tmp_test_directory, request):
+    """
+    Prepare configuration and command for integration tests.
+
+    Parameters
+    ----------
+    config: dict
+        Configuration dictionary.
+    tmp_test_directory: str
+        Temporary test directory (from pytest fixture).
+    request: request
+        Request object.
+
+    Returns
+    -------
+    str: command to run the application test.
+    str: config file model version.
+
+    """
+    tmp_output_path = create_tmp_output_path(tmp_test_directory, config)
+    model_version_requested = request.config.getoption("--model_version")
+
+    if "CONFIGURATION" in config:
+        _skip_test_for_model_version(config, model_version_requested)
+
+        config_file, config_string, config_file_model_version = _prepare_configuration(
+            config["CONFIGURATION"],
+            output_path=tmp_output_path,
+            model_version=model_version_requested,
+        )
+    else:
+        config_file = None
+        config_string = None
+        config_file_model_version = None
+
+    cmd = get_application_command(
+        app=config.get("APPLICATION", None),
+        config_file=config_file,
+        config_string=config_string,
+    )
+    return cmd, config_file_model_version
+
+
+def _skip_test_for_model_version(config, model_version_requested):
+    """Skip test if model version requested is not supported."""
+    if config.get("MODEL_VERSION_USE_CURRENT") is None or model_version_requested is None:
+        return
+    model_version_config = config["CONFIGURATION"]["MODEL_VERSION"]
+    if model_version_requested != model_version_config:
+        pytest.skip(
+            f"Model version requested {model_version_requested} not supported for this test"
+        )
+
+
+def _prepare_configuration(config, output_path, model_version=None):
+    """
+    Prepare test configuration.
```

Comment:
Naming and desciption is very similar to above function `configuration`. Suggest to rename it and also to clarify what `configuration string` means in the docstring.

Reply:
Renamed it to `_prepare_test_options`. Improved also the docstring.